static/app/views/automations/components/ruleNode.tsx

```diff
@@ -0,0 +1,91 @@
+import {Fragment} from 'react';
+import styled from '@emotion/styled';
+
+import {Button} from 'sentry/components/core/button';
+import {IconDelete} from 'sentry/icons';
+import {t} from 'sentry/locale';
+import {space} from 'sentry/styles/space';
+import type {DataCondition} from 'sentry/types/workflowEngine/dataConditions';
+import {
+  DataConditionNodeContext,
+  dataConditionNodesMap,
+  useDataConditionNodeContext,
+} from 'sentry/views/automations/components/dataConditionNodes';
+
+interface RuleNodeProps {
+  condition: Omit<DataCondition, 'condition_group' | 'type' | 'id'>;
+  condition_id: string;
+  onDelete: () => void;
+  onUpdate: (comparison: Record<string, any>) => void;
+}
+
+function Node() {
+  const {condition} = useDataConditionNodeContext();
+  const node = dataConditionNodesMap[condition.comparison_type];
+
+  if (node?.configNode) {
+    return node.configNode();
+  }
+
+  return <Fragment>{node?.label}</Fragment>;
```

Comment:
Can be simplified to
```suggestion
  const Component = node?.configNode;
  return Component ? <Component /> : node?.label;
```